### ISSUE TITLE
Optimize Collection delete

### ISSUE THREAD

**Issue by amityahav** [amityahav/vectory#2]
we need to be able to delete a collection. this can be done by creating a method for the db struct
`DeleteCollection(name string)`. it will delete all the corresponding metadata and existing files.

note: there are subtle edge cases where one thread can potentially use the collection when another deletes it so we want to avoid such case.

**Comment by grumpyp** [comment]
hi there, 

isn't that already implemented?

```
// DeleteCollection deletes collection both on disk and memory.
func (db *DB) DeleteCollection(ctx context.Context, name string) error {
	db.mu.Lock()
	defer db.mu.Unlock()

	if db.closed {
		return ErrDatabaseClosed
	}

	if _, ok := db.collections[name]; !ok {
		return fmt.Errorf("%w: %s", ErrValidationFailed, ErrCollectionDoesntExist)
	}

	err := db.metadataManager.DeleteCollection(ctx, name)
	if err != nil {
		return err
	}

	delete(db.collections, name)

	return nil
}
```

**Comment by amityahav** [comment]
Yes you are right, I might have been totally drunk when writing this. Anyways the edge case where a client still have a reference to the collection when he it is being deleted still exists and needs to be addressed in a way that we prevent the referenced collection from doing anything since it's deleted
